status/status.go

```diff
@@ -114,6 +115,16 @@ func ForLoadTest(test *grpcv1.LoadTest, pods []*corev1.Pod) grpcv1.LoadTestStatu
 		status.StartTime = test.Status.StartTime
 	}
 
+	timeout := time.Duration(test.Spec.TimeoutSeconds) * time.Second
+
+	// Here marked the LoadTest running too long as errored. This status update
+	// could trigger cleanup_agent to terminate its workers.
+	if time.Now().Sub(status.StartTime.Time) >= timeout {
+		status.StopTime = optional.CurrentTimePtr()
+		status.State = grpcv1.Errored
```

Comment:
Great! We should add a reason string and message explaining the timeout. You can see how I did this in #92, adding KubernetesError reason and a string message. 

The reason could be something like `TimeoutExceeded`.